starlite/datastructures/headers.py

```diff
@@ -1,15 +1,396 @@
+"""The code for `MutableHeaders` and parts of `Headers` was adopted from https:
```

Comment:
why do we need to adopt these? why not implement our own version using the new mutldict package?